qiskit/circuit/controlflow/builder.py

```diff
@@ -292,6 +292,9 @@ def assemble(self):
         raise CircuitError("Cannot assemble a placeholder instruction.")
 
     def qasm(self):
+        """
+        Raise an error when trying to convert a placeholder instruction to OpenQASM 2
+        """
         raise CircuitError("Cannot convert a placeholder instruction to OpenQASM 2")
```

Comment:
This whole function wants removing as well - it's just an override of `Instruction.qasm` to catch internal logic errors.  It's not decorated as "deprecated" because instances of these classes should never be accessible publicly.

Reply:
Sounds good to me, I removed the qasm method. Thanks for reviewing! :-)